src/transformation/visitors/access.ts

```diff
@@ -108,10 +109,19 @@ export function transformPropertyAccessExpressionWithCapture(
     node: ts.PropertyAccessExpression,
     thisValueCapture: lua.Identifier | undefined
 ): ExpressionWithThisValue {
-    const property = node.name.text;
     const type = context.checker.getTypeAtLocation(node.expression);
     const isOptionalLeft = isOptionalContinuation(node.expression);
 
+    // FIXME: Stuff like very.nested.call won't work, because
```

Comment:
I tested with 
```ts
/** @noSelf */
interface Nested {
    /** @customName pass */
    fail(): void;
}

declare namespace TestNamespace {
    const nested: Nested;
}

TestNamespace.nested.fail();
```
And
```ts
/** @noSelf */
interface Nested {
    /** @customName pass */
    fail(): void;
}

declare const TestObject: {
    nested: Nested;
};

TestObject.nested.fail();
```

And they both seem to work fine with custom name. Can you give an example  test case where it doesn't work?

Reply:
Sure
```ts
/** @noSelf */
declare namespace TestNamespace {
    /** @customName very.nested.call */
    function fail(): void;
}

TestNamespace.fail();
```

Will output:
```lua
TestNamespace["very.nested.call"]()
```

Instead of:
```lua
TestNamespace.very.nested.call()
```

Reply:
`/** @customName very.nested.call */` Is not really intended behavior for customName at the moment.

I would prefer if you remove this FIXME comment and just deliver this code as it does seem to resolve the original bug. I think allowing nested customName values might be a nice follow-up feature for a separate PR.

Reply:
Sure thing